docs/userguide.md

```diff
@@ -296,6 +301,112 @@ kubefedctl disable <FederatedTypeConfig Name> --delete-from-api
+federated resource.  This may not be compatible with a kubefed control plane that has enabled 
+a federated type in a non-default way (e.g. the group of the federated type has been set to 
+something other than `types.kubefed.k8s.io`).
+
+### Federate a namespace 
+Federating a namespace is a special case. Although the mechanism to federate the namespace
+using `kubefedctl federate` is same, kubefed treats the target namespace in a peculiar way. 
+On federating a target namespace from a host cluster, kubefed would use the same target namespace 
+of the host cluster as the container for the federated namespace resource. 
+
+Using this strategy allows for a very helpful use case of [Namespaced Federation](#namespaced-federation), 
+where the whole kubefed API and kubefed controllers can be deployed in a single target namespace. 
+This also simplifies the concept of the container for the federated resources. A kubernetes namespace 
+in the host cluster is the container for all federated resources including the federated namespace.
+This also means that any target resource can be federated without the federated namespace
+being in place. The container for the federated resources which is the kubernetes namespace 
+should be in place.
+
+Another peculiar point to note is that if the namespaced resources are federated before federating
+the target namespace, they won't be propagated to the member clusters. This is because the sync
+controller computes the placement of resources considering the placement of the federated
+namespace and the placement of the resource itself. The final placement decision is the intersection
+of both and if the federated namespace is missing, the resulting placement decision is nil. 
+
+***Example:***
+Federate a namespace named "myNs" 
+```bash
+kubefedctl federate namespace myNs
+```
+
+### Federate a namespace with contents
+`kubefedctl federate` can also be used to federate a target namespace and all its contained resources
+in one shot. This can be achieved using the flag `--contents` which is valid only when the `<target 
+kubernetes API type>` is a `namespace`. `kubefedctl federate` with `--contents` looks up all the 
+existing resources in the target namespace and federates them one by one. It however skips the
+controller managed resources viz. the `endpoints` and `events`.
+Similar to controller managed resources user might want to additionally skip resources from being 
+federated. `--skip-api-resources` can be used to skip such resources. 
+
+***Example:***
+Federate a namespace named "myNs" skipping API Resource "configmaps" and API Resource group "apps" 
+```bash
+kubefedctl federate namespace myNs --contents --skip-api-resources "configmaps,apps"
+```
+
+### Optionally enable type while federating a resource
+`kubefedctl federate` allows optionally enabling the given `<target kubernetes API type>` before
+federating the resource. Flag `--enable-type` can be used to achieve this. Using this will 
```

Comment:
`s/federating the resource.*achieve this/federating the resource by supplying the `--enable-type` flag./`
`s/Using this will.*/This will enable federation of the target type if it is not already enabled/`

Reply:
Updated!